aptos-move/framework/aptos-framework/sources/account.move

```diff
@@ -838,16 +859,41 @@ module aptos_framework::account {
     public entry fun test_valid_rotate_authentication_key_multi_ed25519_to_ed25519(account: signer) acquires Account, OriginatingAddress {
         initialize(&account);
 
-        let curr_pk = multi_ed25519::new_unvalidated_public_key_from_bytes(MULTI_ED25519_SRC_PK_BYTES);
-        let curr_auth_key = multi_ed25519::unvalidated_public_key_to_authentication_key(&curr_pk);
+        let (curr_sk, curr_pk) = multi_ed25519::generate_keys(3, 2);
+        let curr_pk_unvalidated = multi_ed25519::public_key_to_unvalidated(&curr_pk);
+        let curr_auth_key = multi_ed25519::unvalidated_public_key_to_authentication_key(&curr_pk_unvalidated);
         let alice_address = from_bcs::to_address(curr_auth_key);
         let alice = create_account_unchecked(alice_address);
 
-        rotate_authentication_key(&alice, MULTI_ED25519_SCHEME, MULTI_ED25519_SRC_PK_BYTES, ED25519_SCHEME, ED25519_TO_PK_BYTES, SIG_CAP_ROTATE_KEY_MULTI_TO_SINGLE, SIG_CAP_UPDATE_TABLE_MULTI_TO_SINGLE);
-        let address_map = &mut borrow_global_mut<OriginatingAddress>(@aptos_framework).address_map;
-        let new_pk = ed25519::new_unvalidated_public_key_from_bytes(ED25519_TO_PK_BYTES);
-        let new_auth_key = ed25519::unvalidated_public_key_to_authentication_key(&new_pk);
+        let addr = signer::address_of(&alice);
```

Comment:
This code feels repetitive now: I think there was similar code in the previous test you changed. Is it worth wrapping in a `#[test-only]` function.

Reply:
IMO it's ok if we don't wrap them here. Actually It's a bit better for readers to see the data derivation path without jumping to another function.